libsql/src/statement.rs

```diff
@@ -92,18 +93,37 @@ pub struct MappedRows<F> {
     map: F,
 }
 
-impl<F, T> Iterator for MappedRows<F>
+impl<F> MappedRows<F> {
+    pub fn new(rows: Rows, map: F) -> Self {
+        Self { rows, map }
+    }
+}
+
+#[cfg(feature = "core")]
+impl<F, T> futures::Stream for MappedRows<F>
 where
-    F: FnMut(Row) -> Result<T>,
+    F: FnMut(Row) -> Result<T> + Unpin,
 {
     type Item = Result<T>;
 
-    #[inline]
-    fn next(&mut self) -> Option<Self::Item> {
-        let map = &mut self.map;
-        self.rows
-            .next()
-            .transpose()
-            .map(|row_result| row_result.and_then(map))
+    fn poll_next(
+        mut self: std::pin::Pin<&mut Self>,
+        cx: &mut Context<'_>,
+    ) -> Poll<Option<Self::Item>> {
+        use futures::{ready, Future};
+
+        let mut rows = unsafe { self.as_mut().map_unchecked_mut(|pin| &mut pin.rows) };
+        let mut fut = Box::pin(rows.next());
+        let res = ready!(fut.as_mut().poll(cx));
```

Comment:
I don' think this is correct, if the rows future is not available right away and returns pending we cancel it. What happens there? I think we need to store the future inside the stream itself.

Reply:
If I got this correctly, this type works essentially like iter Map. I modified it into most straightforward way possible. I don't think it should have a freedom of cancelling the underlying futures on its own. Maybe there's some other Stream-based mapper that we could use instead of this one?